rust/windows-client/src-tauri/src/client/gui.rs

```diff
@@ -21,6 +21,12 @@ use ControllerRequest as Req;
 
 mod system_tray_menu;
 
+/// The Windows client doesn't use platform APIs to detect network connectivity changes,
```

Comment:
@ReactorScram I'm not sure if this is accurate anymore, but for example on the Android client this actually works quite well coupled with the `MAX_INTERVAL` of 5s set above.

Doing it with connlib like this here has the downside that the client could lose connectivity for a few seconds without communicating anything to the user, but the upside that it's simpler and has the same behavior across platforms.

Doing it via the platform APIs has the upside that we can communicate to the user and reconnect slightly more quickly but we need a state machine with an in-between state of "signed in but disconnected" state in the UI.

Reply:
We could keep using platform APIs on platforms that support them if you've already built this in the Windows client, but then fallback to this implementation within connlib if not.

Reply:
@jamilbk It's still pretty accurate. #3262 will log the network changes but it won't use them to reconnect connlib, because of #2975.

And yeah the Windows API seems pretty reliable, it can tell us within about 3 seconds when we gain or lose Internet connectivity. I don't have any middle states for LAN-only or IPv4-only but those could be added.

So if connlib can take those as a hint like "Hey the Internet is probably back" it would be cool. Otherwise I can block it out or just spin it off as a utility crate and delete it from the code if we don't need it.

Reply:
Sounds good. After discussion today it seems like we probably need both implementations anyway.